### PR TITLE
PLT-43: migrate mock node out of the PAB

### PR DESCRIPTION
**Pull request by sjoerdvisscher**
Pre-submit checklist:
- Branch
    - [ ] Tests are provided (if possible)
    - [ ] Commit sequence broadly makes sense
    - [ ] Key commits have useful messages
    - [ ] Formatting, PNG optimization, etc. are updated
    - [ ] Important changes are reflected in changelog.d of the affected packages
- PR
    - [ ] Self-reviewed the diff
    - [ ] Useful pull request description
    - [ ] Reference the ADR in the PR and reference the PR in the ADR (if revelant)
    - [ ] Reviewer requested

### CODE REVIEW

File: cardano-node-socket-emulator/src/Cardano/Node/Socket/Emulator/Types.hs [hunk 1]
```diff
@@ -0,0 +1,154 @@
...
+initialChainState =
+    fromEmulatorChainState . view EM.chainState . EM.emulatorState . pure . pure . Valid .
+    createGenesisTransaction . Map.mapKeys EM.mockWalletAddress
+
+
+-- Logging ------------------------------------------------------------------------------------------------------------
+
+-- | Top-level logging data type for structural logging
+-- inside the PAB server.
+data PABServerLogMsg =
```
Comment (koslambrou) [inline cardano-node-socket-emulator/src/Cardano/Node/Socket/Emulator/Types.hs:141]: Remove PAB reference?

File: cardano-node-socket-emulator/src/Cardano/Node/Socket/Emulator/Types.hs [hunk 2]
```diff
@@ -0,0 +1,154 @@
...
+        , nscProtocolParametersJsonPath :: Maybe FilePath
+        -- ^ Path to a JSON file containing the protocol parameters
+        }
+    deriving stock (Show, Eq, Generic)
+    deriving anyclass (FromJSON, ToJSON)
+
+defaultNodeServerConfig :: NodeServerConfig
+defaultNodeServerConfig =
+    NodeServerConfig
+      -- See Note [pab-ports] in 'test/full/Plutus/PAB/CliSpec.hs'.
```
Comment (koslambrou) [inline cardano-node-socket-emulator/src/Cardano/Node/Socket/Emulator/Types.hs:69]: Is this still relevant?

File: cardano-node-socket-emulator/src/Cardano/Node/Socket/Emulator/Types.hs [hunk 3]
```diff
@@ -0,0 +1,154 @@
...
+    createGenesisTransaction . Map.mapKeys EM.mockWalletAddress
+
+
+-- Logging ------------------------------------------------------------------------------------------------------------
+
+-- | Top-level logging data type for structural logging
+-- inside the PAB server.
+data PABServerLogMsg =
+    StartingSlotCoordination UTCTime Millisecond
+    | StartingPABServer Int
```
Comment (koslambrou) [inline cardano-node-socket-emulator/src/Cardano/Node/Socket/Emulator/Types.hs:143]: Same here.

**Review by koslambrou — APPROVED** [review]
LGTM :)

File: cardano-node-socket-emulator/src/Cardano/Node/Socket/Emulator/Types.hs (hunk 2 above)
Comment (sjoerdvisscher) [reply cardano-node-socket-emulator/src/Cardano/Node/Socket/Emulator/Types.hs:69]: Unfortunately, yes.

**Pull request merged**